theseus/core/objective.py

```diff
@@ -532,11 +538,14 @@ def update_vectorization_if_needed(self):
     def __iter__(self):
         return iter([cf for cf in self.cost_functions.values()])
 
-    def _get_iterator(self):
+    def _get_error_iter_base(self) -> Iterable:
+        return iter(cf for cf in self.cost_functions.values())
+
+    def _get_jacobians_iter(self) -> Iterable:
```

Comment:
Trying to understand the difference between error iter and jacobians iter since they both return on cost_functions.values().

Reply:
The ones that return `self.cost_functions.values()` are the non-vectorized versions, which should be the same in both cases. When vectorized is on, they return different iterators: the jacobians iter will point to containers that have a permanent cache of errors and jacobians, the error iter will point to a tmp container that only stores errors computed on demand. I added some comments at the top explaining `_get_error_iter`, but maybe the variable names are also non-descriptive enough.

Reply:
Okay that makes sense. Thanks for the explanation and comment will be useful.